contracts/interchain-token/src/contract.rs

```diff
@@ -37,7 +37,13 @@ impl InterchainToken {
         Self::write_metadata(&env, token_meta_data);
 
         env.storage().instance().set(&DataKey::TokenId, &token_id);
-        env.storage().instance().set(&DataKey::Minter(minter), &());
+
+        if let Some(minter_address) = minter {
+            env.storage()
+                .instance()
+                .set(&DataKey::Minter(minter_address), &());
+        }
```

Comment:
```suggestion
        if let Some(minter) = minter {
            env.storage()
                .instance()
                .set(&DataKey::Minter(minter), &());
        }
```
could also use `minter.and_then(...)`

Reply:
updated